### PR TITLE
horizontal scroll bar with sticky footer bug fix [ ScrollablePane ]

### LINKED ISSUE #6339: Horizontal scrollbar is hidden behind Sticky footer
### Bug Report 
 
 
- __Package version(s)__: 6.53.0 onwards
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No)
 
Requested priority: (Blocking, High, Normal, Low) 
 Normal
Products/sites affected: (if applicable)
 
#### Describe the issue:
 
The ScrollablePane adjusts the position of the Sticky footer(s) based on the presence of the horizontal scrollbar. The presence of the scrollbar is checked on various events like window resize etc. But it is not checked based on horizontally expanding content. The horizontal scrollbar shows up if the content overflows horizontally, but is hidden behind the Sticky footer. 

Following steps reproduces this on the ScrollbablePane DetailsList example -

* Ensure that the horizontal scrollbar doesnt exist (reduce the number of colums for this or use a large monitor)
* Resize a column in 'Item List' such that the content overflows (notice the height of the vertical scrollbar as you resize)
 
#### Actual behavior:
 
The horizontal scrollbar is hidden behind the sticky footer. It shows up once you scroll vertically or resize window etc.
 
#### Expected behavior:
 
The sticky footer should move when the horizontal scrollbar is visible.

### PR DESCRIPTION
**Pull request by arkogupta**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #6339 
- [X] Include a change request file using `$ npm run change`

#### Description of changes
Updates ScrollablePaneBase component by making changes to the componentDidMount lifecycle hook. The MutationObserver has been modified by adding a check which compares the current computed scroll bar height and comparing it with the height in the current state. If the height has changed we explicitly update the state. This re-renders the scroll bar and hence it is positioned correctly below the footer.

This fix, as compared to the [previous one ](https://github.com/OfficeDev/office-ui-fabric-react/pull/6666) by @mikewheaton seems to be more efficient as it doesn't involve re-rendering the scrollablePane( and hence all it's children ) on each column-resize operation.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx [hunk 1]
```diff
@@ -109,6 +109,13 @@ export class ScrollablePaneBase extends BaseComponent<IScrollablePaneProps, IScr
           return false;
         }
 
+        const scrollbarHeight = this._getScrollbarHeight();
```
Comment (shagra-ms) [inline packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx]: Add a comment here explaining why below is required.

**Comment by KevinTCoughlin** [comment]
I'm confused by

> This fix, as compared to the [previous one ](https://github.com/OfficeDev/office-ui-fabric-react/pull/6666) by @mikewheaton seems to be more efficient as it doesn't involve re-rendering the scrollablePane( and hence all it's children ) on each column-resize operation.

Two questions:

- Can you please explain a bit more regarding this change? Does it avoid the call added in the above PR due to the usage of `componentDidMount`? Perhaps browser profile snapshots would be useful here.
- Also is this behavior that I can visibly notice? If so will check the deploy links / checkout.

I'd like to sign-off on the PR but I think I require more context to do so.

**Comment by arkogupta** [comment]
Talking strictly about performance, we observed that through both the approaches there is not much noticeable difference. But there are two points which are of significance which are in favour of this approach:

- We are letting the component re-render only after checking if the scroll bar height has changed or not( i.e if the scroll bar has appeared/disappeared due to change of width of the content ). On the other hand, the previous fix caused the scrollablePane to re-render on each column-resize operation irrespective of whether overflow occurred or not. 

- There is no dependency for the consumer to make any callbacks whatsoever.

**Comment by arkogupta** [comment]
> **Edit:** After rereading, it seems like this PR might _replace_ the fix introduced in #6666?

Yes Kevin, that is what is intended.

**Comment by cliffkoh** [comment]
Pinged @mikewheaton .

**Comment by Jahnp** [comment]
@mikewheaton and @leddie24 , would you guys mind taking a look here? Thanks!

**Pull request merged**